lib/model/indexhandler.go

```diff
@@ -111,20 +111,42 @@ func newIndexHandler(conn protocol.Connection, downloads *deviceDownloadState, f
 	}
 }
 
+// waitForFileset waits for the handler to resume and fetches the current fileset.
+func (s *indexHandler) waitForFileset(ctx context.Context) *db.FileSet {
+	s.cond.L.Lock()
+	defer s.cond.L.Unlock()
+
+	for s.paused && ctx.Err() == nil {
+		s.cond.Wait()
+	}
+	fset := s.fset
+
+	return fset
```

Comment:
This returns a `nil` fileset if the context is cancelled, without the caller knowing about it.